src/core/api_client_parallel.py

```
@@ -1,116 +1,155 @@
     async def get_many(self, requests: List[Tuple[str, Optional[Dict]]]) -> List[Dict]:
         """Make multiple requests in parallel using all available keys"""
         results = []
-        
-        async with aiohttp.ClientSession() as session:
+
+        connector = aiohttp.TCPConnector(limit=100, limit_per_host=30)
+        async with aiohttp.ClientSession(connector=connector) as session:
             # Create tasks for all requests
             tasks = []
+            # Use random key assignment to better distribute load
+            import random
```

Ruff (0.12.2)

`random` imported but unused

Remove unused import: `random`

(F401)